plugins/modules/vmware_content_deploy_ovf_template.py

```diff
@@ -0,0 +1,236 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2020, Lev Goncharov <lev@goncharov.xyz>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = r'''
+---
+module: vmware_content_deploy_ovf_template
+short_description: Deploy Virtual Machine from ovf template stored in content library.
+description:
+- Module to deploy virtual machine from ovf template in content library.
+- All variables and VMware object names are case sensitive.
+version_added: '2.9'
+author:
+- Lev Goncharv (@ultral)
+notes:
+- Tested on vSphere 6.7
+requirements:
+- python >= 2.6
+- PyVmomi
+- vSphere Automation SDK
+options:
+    ovf_template:
+      description:
+      - The name of OVF template from which VM to be deployed.
+      type: str
+      required: True
+      aliases: ['ovf','template_src']
+    name:
+      description:
+      - The name of the VM to be deployed.
+      type: str
+      required: True
+      aliases: ['vm_name']
+    datacenter:
+      description:
+      - Name of the datacenter, where VM to be deployed.
+      type: str
+      required: True
+    datastore:
+      description:
+      - Name of the datastore to store deployed VM and disk.
+      type: str
+      required: True
+    folder:
+      description:
+      - Name of the folder in datacenter in which to place deployed VM.
+      type: str
+      required: True
+    host:
+      description:
+      - Name of the ESX Host in datacenter in which to place deployed VM.
+      type: str
+      required: True
+    resource_pool:
+      description:
+      - Name of the resourcepool in datacenter in which to place deployed VM.
+      type: str
+      required: False
+    cluster:
+      description:
+      - Name of the cluster in datacenter in which to place deployed VM.
+      type: str
+      required: False
+extends_documentation_fragment: vmware_rest_client.documentation
```

Comment:
Do you think we should add a parameter called "state" like "present" and "poweredon"? I am not sure if the OVF deploy API provides the same (just like template deploy API). Not a blocker though.

Reply:
As I can see there are no options like that http://vmware.github.io/vsphere-automation-sdk-rest/6.5/operations/com/vmware/vcenter/ovf/library_item.deploy-operation.html . On the one hand it might be good idea to unify modules behavior, but on the over hand we will put to much complexity into the module. I prefer to don't introduce state parameter. In other words I'd like to have 1 simple module for 1 API method.

Reply:
Cool. I totally agree with you.